File: dotnet/KeepScreenAwake.Core/Transport/Protocol.cs
using KeepScreenAwake.Core.Config;

namespace KeepScreenAwake.Core.Transport;

public enum CommandType { Status, On, Off, Mode, Logs }

public class Request
{
    public CommandType Command { get; set; }
    public string? Mode { get; set; } // for Mode command
    public int Lines { get; set; } = 50;
}

public class Response
{
    public bool Ok { get; set; }
    public object? Data { get; set; }
    public string? Error { get; set; }
}

public class StatusData
{
    public string Mode { get; set; } = "";
    public bool AwakeActive { get; set; }
    public bool DisplayOnly { get; set; }
    public List<ScheduleWindowData> Schedule { get; set; } = new();
}

public class ScheduleWindowData
{
    public string Start { get; set; } = "";
    public string End { get; set; } = "";
    public List<string> Days { get; set; } = new();
}

public class LogsData
{
    public List<string> Lines { get; set; } = new();
}

File: go/internal/transport/protocol.go
package transport

// Command identifies the IPC operation requested by the CLI.
type Command string

const (
	CmdStatus Command = "status"
	CmdOn     Command = "on"
	CmdOff    Command = "off"
	CmdMode   Command = "mode"
	CmdLogs   Command = "logs"
)

// Request is the JSON envelope sent from CLI to daemon.
type Request struct {
	Command Command `json:"command"`
	Mode    string  `json:"mode,omitempty"`
	Lines   int     `json:"lines,omitempty"`
}

// Response is the JSON envelope returned by the daemon.
type Response struct {
	OK    bool   `json:"ok"`
	Data  any    `json:"data,omitempty"`
	Error string `json:"error,omitempty"`
}

// StatusData is the payload for a status response.
type StatusData struct {
	Mode        string   `json:"mode"`
	AwakeActive bool     `json:"awake_active"`
	DisplayOnly bool     `json:"display_only"`
	Schedule    []Window `json:"schedule,omitempty"`
}

// Window mirrors config.Window for transport use.
type Window struct {
	Start string   `json:"start"`
	End   string   `json:"end"`
	Days  []string `json:"days"`
}

// LogsData is the payload for a logs response.
type LogsData struct {
	Lines []string `json:"lines"`
}
